packages/read-model/src/Product/Listed/ListedProductView.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Product\Listed;
+
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+use Shopsys\ReadModelBundle\Image\ImageViewInterface;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionView;
+use Shopsys\ReadModelBundle\Product\Action\ProductActionViewInterface;
+
+/**
+ * @experimental
+ *
+ * Class representing products in lists in FE templates (to avoid usage of Doctrine entities a hence achieve performance gain)
+ */
+class ListedProductView implements ListedProductViewInterface
+{
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Image\ImageViewInterface|null
+     */
+    protected $image;
+
+    /**
+     * @var string
+     */
+    protected $availability;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice
+     */
+    protected $sellingPrice;
+
+    /**
+     * @var string|null
+     */
+    protected $shortDescription;
+
+    /**
+     * @var int[]
+     */
+    protected $flagIds = [];
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Product\Action\ProductActionView
+     */
+    protected $action;
+
+    /**
+     * ListedProductView constructor.
+     * @param int $id
+     * @param string $name
+     * @param string|null $shortDescription
+     * @param string $availability
+     * @param \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice $sellingPrice
+     * @param int[] $flagIds
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionViewInterface $action
+     * @param \Shopsys\ReadModelBundle\Image\ImageViewInterface|null $image
+     */
+    public function __construct(
+        int $id,
+        string $name,
+        ?string $shortDescription,
+        string $availability,
+        ProductPrice $sellingPrice,
+        array $flagIds,
+        ProductActionViewInterface $action,
+        ?ImageViewInterface $image
+    ) {
+        $this->validateFlagIds($flagIds);
```

Comment:
I just saw even nicer solution!

```suggestion
        \Webmozart\Assert\Assert::allInteger($flagIds);
```

Reply:
@grossmannmartin what do you think about this solution?

Reply:
I like the idea of "centralized assertion" (no need to invent it everywhere), but I wasn't sure if we really need this. Discussion with @PetrHeinz, you and some deep internet digging leave me with a very good feeling about it.
So I added the package to composer (as we didn't had it directly) and apply this suggestion.

Reply:
I will reorder commits then so it will make sense :) I didn't want to do "used webmozart" commit

Reply:
:+1: